### ISSUE TITLE
Add a ton of links :)

### ISSUE THREAD

**Issue by killercup** [stevedonovan/gentle-intro#16]
Very nice, I really like your style of writing! It's a bit more _sudden_ in introducing new things, and mentions a lot of nice details on the side. I only wish you'd add some links to learn more about that :)

For example, [here][1], when you introduce formatting syntax with `println!("{:.1} ",x);`, people might wonder where they can lear more about this, so why not add a link to the `std::fmt` docs? (This is also one of the most underappreciated parts of the std docs!)

Or, when you introduce Option, Vec, Iterator, etc. and present a few methods – the API docs have _so many_ other interesting things!

[1]: https://github.com/stevedonovan/gentle-intro/blob/9f9730aa2c3c3d678dc76d1055562df50cad678c/2-structs-enums-lifetimes.md#example-iterator-over-floating-point-range

**Comment by stevedonovan** [comment]
You're quite right - just saying 'like Python' is a bit throwaway. Yeah, the more links the better - and there are some brilliant sections of the documentation like `std::fmt`.  I suspect I 'underwrite' a bit, need to flesh out those interesting details.

**Comment by torkleyy** [comment]
Links between the pages would be nice, too.

**Comment by stevedonovan** [comment]
That's one of those ideas which are obvious in hindsight. It's not really a book and I can refer back to earlier sections.

Another thing I'm thinking of, is providing links to the source. The `src` directory has a lot of little test programs, and I may be able to persuade mdbook to render them for me.

As I go through the text, I'm putting in as many doc links as possible. There's even a section on how to read the docs (start of section 5) because man the first time I saw them I was scared (and I don't scare easily)
